File: modal/src/main/java/ir/drax/modal/model/MoButton.java
package ir.drax.modal.model;

import android.view.View;

public class MoButton {
    private CharSequence text;
    private int icon;
    private OnClickListener clickListener;

    public MoButton(CharSequence text, int icon, OnClickListener clickListener) {
        this.text = text;
        this.icon = icon;
        this.clickListener = clickListener;
    }

    public CharSequence getText() {
        return text;
    }

    public int getIcon() {
        return icon;
    }

    public OnClickListener getClickListener() {
        return clickListener;
    }

    public interface  OnClickListener{
        public boolean onClick(View v);
    }
}

File: modal/src/main/java/ir/drax/modal/model/MoButton.kt
package ir.drax.modal.model

import android.view.View

typealias MoClickListener = (button: View) -> Boolean
open class MoButton(val displayText:CharSequence="", val iconResourceId:Int=0, val clickListener:MoClickListener={ true })
class JvmMoButton(val text:CharSequence="", val icon:Int=0,callBack:OnClickListener?=null):MoButton(text,icon,{ callBack?.onClick(null)?:true })

interface OnClickListener{
    fun onClick(view: View?):Boolean
}